datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+    hence can be used to ship (default and branch-specific) configuration with
+    a dataset.
+
+    Any DATALAD_* environment variable is also mapped to a configuration item.
+    Their values take precedence over any other specification. In variable
+    names '_' encodes a '.' in the configuration name, and '__' encodes a '-',
+    such that 'DATALAD_SOME__VAR' is mapped to 'datalad.some-var'.
+
+    Recursive operation is supported for querying and modifying configuration
+    across a hierarchy of datasets.
+    """
+    _examples_ = [
+        dict(text="Dump the effective configuration, including an annotation for common items",
+             code_py="configuration()",
+             code_cmd="datalad configuration"),
+        dict(text="Query two configuration items",
+             code_py="configuration('get', ['user.name', 'user.email'])",
+             code_cmd="datalad configuration get user.name user.email"),
+        dict(text="Recursively set configuration in all (sub)dataset repositories",
+             code_py="configuration('set', [('my.config.name', 'value')], recursive=True)",
+             code_cmd="datalad configuration -r set my.config=value"),
+        dict(text="Modify the persistent dataset configuration (changes are not committed)",
+             code_py="configuration('set', [('my.config.name', 'value')], scope='dataset')",
+             code_cmd="datalad configuration --scope dataset set my.config=value"),
+    ]
+
+    result_renderer = 'tailored'
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to query or to configure""",
+            constraints=EnsureDataset() | EnsureNone()),
+        action=Parameter(
+            args=("action",),
+            nargs='?',
+            doc="""which action to perform""",
+            constraints=EnsureChoice(*config_actions)),
+        scope=Parameter(
+            args=("--scope",),
+            doc="""identify to configuration scope for getting or setting
+            configuration. If no scope is declared for a query, all
+            configuration sources (including overrides via environment
+            variables) are considered according to the normal
+            rules of precedence. For action 'get' only 'dataset' and 'local'
+            (with include 'global' here) are supported. For action 'dump',
+            a scope selection is ignored and all scopes are considered.""",
+            constraints=EnsureChoice('global', 'local', 'dataset', None)),
+        spec=Parameter(
+            args=("spec",),
+            doc="""configuration name (for actions 'get' and 'unset'),
+            or name/value pair (for action 'set')""",
+            nargs='*',
+            metavar='name[=value]',
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+    )
+
+    @staticmethod
+    @datasetmethod(name='cfg')
```

Comment:
I'm guessing this is a stale part from a previous iteration and meant to match the unbound name, `configuration`.

Reply:
Fixed. Thx!